src/cephadm/cephadm.py

```diff
@@ -3265,12 +3408,19 @@ def get_container(ctx: CephadmContext,
         name = '%s.%s' % (daemon_type, daemon_id)
         envs.extend(Keepalived.get_container_envs())
         container_args.extend(['--cap-add=NET_ADMIN', '--cap-add=NET_RAW'])
+    elif daemon_type == CephNvmeof.daemon_type:
+        name = '%s.%s' % (daemon_type, daemon_id)
+        container_args.extend(['--ulimit', 'memlock=-1:-1']) # TODO(redo): FIXME
+        container_args.extend(['--cap-add=SYS_ADMIN', '--cap-add=CAP_SYS_NICE'])
     elif daemon_type == CephIscsi.daemon_type:
         entrypoint = CephIscsi.entrypoint
         name = '%s.%s' % (daemon_type, daemon_id)
         # So the container can modprobe iscsi_target_mod and have write perms
         # to configfs we need to make this a privileged container.
         privileged = True
+    elif daemon_type == CephNvmeof.daemon_type:
```

Comment:
Hello @rkachach 🖖
Code in lines 3421-3423 appears unreachable, since the same condition ` elif daemon_type == CephNvmeof.daemon_type:` appears in line 3411 in the same elif chain.

Reply:
@baum good catch :+1:  that code was not reached/used indeed. Deleting it.